aws/resource_aws_spot_instance_request.go

```diff
@@ -84,6 +86,16 @@ func resourceAwsSpotInstanceRequest() *schema.Resource {
 				Default:  "terminate",
 				ForceNew: true,
 			}
+			s["valid_from"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			}
```

Comment:
Switching to RFC3339 gives us an easy plan-time validation function for both of these:

```go
ValidateFunc: validation.ValidateRFC3339TimeString,
```

Reply:
Added the validation function from the same package(validators.go).